src/main/java/com/autotune/analyzer/performanceProfiles/utils/PerformanceProfileUtil.java

```diff
@@ -0,0 +1,177 @@
+            for (Containers containers : deploymentResultData.getContainers()) {
+                HashMap<AnalyzerConstants.MetricName, HashMap<String, MetricResults>> containerMetricsMap =
+                        containers.getContainer_metrics();
+                List<String> kruizeFunctionVariablesList = containerMetricsMap.keySet().stream().toList().stream().map(Enum::name).toList();
+                LOGGER.debug("perfProfileFunctionVariablesList: {}", perfProfileFunctionVariablesList);
+                LOGGER.debug("kruizeFunctionVariablesList: {}", kruizeFunctionVariablesList);
+                if (!kruizeFunctionVariablesList.containsAll(mandatoryFields)) {
+                    errorMsg = errorMsg.concat(String.format("Missing one of the following mandatory parameters for experiment - %s : %s", experimentResultData.getExperiment_name(), mandatoryFields));
+                    break;
+                }
+                for (Map.Entry<AnalyzerConstants.MetricName, HashMap<String, MetricResults>> funcVar : containerMetricsMap.entrySet()) {
+                    Map<String, Object> aggrInfoClassAsMap;
+                    if (!aggrFunctionsObjects.isEmpty()) {
+                        try {
+                            aggrInfoClassAsMap = convertObjectToMap(funcVar.getValue().get("results").getAggregationInfoResult());
+                            errorMsg = validateAggFunction(aggrInfoClassAsMap.keySet(), aggrFunctionsObjects);
+                            if (!errorMsg.isBlank()) {
+                                errorMsg = errorMsg.concat(String.format("for the experiment : %s"
+                                        , experimentResultData.getExperiment_name()));
+                                break;
+                            }
+                        } catch (IllegalAccessException | InvocationTargetException e) {
+                            throw new RuntimeException(e);
+                        }
+                    } else {
+                        // check if query is also absent
+                        if (queryList.isEmpty()) {
+                            errorMsg = AnalyzerErrorConstants.AutotuneObjectErrors.QUERY_FUNCTION_MISSING;
+                            break;
+                        }
+                    }
+                    // check if the 'value' is present in the result JSON
+                    if (null == funcVar.getValue().get("results").getValue()) {
+                        LOGGER.debug(AnalyzerErrorConstants.AutotuneObjectErrors.MISSING_VALUE.concat(funcVar.getKey().toString()));
+                    }
+                }
+            }
+        }
+        return errorMsg;
+    }
+
+    public static void addPerformanceProfile(Map<String, PerformanceProfile> performanceProfileMap, PerformanceProfile performanceProfile) {
+        performanceProfileMap.put(performanceProfile.getName(), performanceProfile);
+        LOGGER.info("Added PerformanceProfile: {} ",performanceProfile.getName());
+    }
```

Comment:
Why is this in util?

Reply:
I thought it's not related to the impls as the method accepts map and object so it's jus adding it to map. The caller has the scope of both so there is no internal entity in impls interacting with this so I thought it can be moved to utils